packages/dev/loaders/src/BVH/bvhLoader.ts

```diff
@@ -0,0 +1,388 @@
+        }
+        context.numFrames = numFrames;
+
+        // frame time
+        const frameTimeLine = lines.shift();
+        if (!frameTimeLine) {
+            throw new Error("Unexpected end of file before frame time");
+        }
+        const frameTimeTokens = frameTimeLine.trim().split(/[\s]+/);
+        if (frameTimeTokens.length < 3) {
+            throw new Error("Invalid frame time line");
+        }
+        const frameTime = parseFloat(frameTimeTokens[2]);
+        if (isNaN(frameTime)) {
+            throw new Error("Failed to read frame time.");
+        }
+
+        context.frameRate = frameTime;
+
+        // read frame data line by line
+        for (let i = 0; i < numFrames; ++i) {
+            const frameLine = lines.shift();
+            if (!frameLine) {
+                continue;
+            }
+            const tokens = frameLine.trim().split(/[\s]+/);
+            BVHParser._ReadFrameData(tokens, i * frameTime, root);
+        }
+
+        context.root = root;
+
+        this._ConvertNode(context.root, null, context);
+
+        context.skeleton.returnToRest();
+        return context.skeleton;
+    }
+
+    /**
+     * Converts a BVH node to a Babylon bone
+     * @param node - The BVH node to convert
+     * @param parent - The parent bone
+     * @param context - The loader context
+     */
+    private static _ConvertNode(node: BVHNode, parent: Nullable<Bone>, context: LoaderContext) {
+        const matrix = this._BoneOffset(node);
+        const bone = new Bone(node.name, context.skeleton, parent, matrix);
+
+        // Create animation for this bone
+        const animation = this._CreateAnimations(node, context);
+        if (animation) {
+            // Apply rotation correction to the root bone's animation keys
+            if (!parent) {
+                // Check if it's the root node
+                const correctionMatrix = Matrix.RotationAxis(Axis.X, Math.PI / 2); // -90 degrees on X-axis
+                const correctedKeys = animation.getKeys().map((key: IAnimationKey) => {
+                    const originalMatrix = key.value as Matrix;
+                    // Apply correction: We want to rotate the final orientation, so post-multiply
+                    const correctedMatrix = originalMatrix.multiply(correctionMatrix);
+                    return { frame: key.frame, value: correctedMatrix };
+                });
+                animation.setKeys(correctedKeys);
+            }
+            bone.animations.push(animation);
+        }
+
+        for (const child of node.children) {
+            this._ConvertNode(child, bone, context);
+        }
+    }
+
+    /**
+     * Converts the BVH node's offset to a Babylon matrix
+     * @param node - The BVH node to convert
+     * @returns The converted matrix
+     */
+    private static _BoneOffset(node: BVHNode): Matrix {
+        // Convert BVH Y-up, right-handed offset to Babylon's Y-up, left-handed system.
```

Comment:
is rotation expected to be correct?

Reply:
Nice catch! Actually, those comments are wrong, those are left-overs from my attempts to fix the rotation using AI.